oidcclient/jwk.go

```diff
@@ -111,10 +122,16 @@ func (ks *OIDCTenant) getJWKsFromServer(zoneID string) (r interface{}, err error
 	defer resp.Body.Close()
 
 	if resp.StatusCode != http.StatusOK {
-		ks.acceptedZoneIds[zoneID] = false
-		return result, fmt.Errorf("failed to fetch jwks from remote for x-zone_uuid %s: %v (%s)", zoneID, err, resp.Body)
+		ks.acceptedTenants[tenant] = false
```

Comment:
I know it was like this before, but should we maybe add a list of status codes which indicate an invalid token?
Otherwise we might block a token for the whole caching period because of temporary network issues.
@liga-oz How is this done in the Java client libs?

Reply:
we check the status code exactly the same way like here L124 
I don't understand how would you block a token for the whole caching period if only validated tokens would be stored in a security context and if JWKs are not available then the token is invalid anyway? (at least in java)

Reply:
I thought of the following scenario: A user calls the application with a valid JWT token for the first time (with its clientID, appTID, Azp combination), so the client lib requests the JWKs from the server. For whatever reason there is e.g. a network issue, the request ist not successful and returns 503 for this single call. The next call would return 200 as expected. However the client lib stored the tenant with this clientID, appTID, Azp combination as not accepted (line 125) because of the previous 503. This will persist until the caching period is expired and the client lib in any case fetches new JWKs from the server.

Reply:
OK, I see what you mean. In java lib we do not store unsuccessful requests to the JWKs endpoint or the client header combination, if the response is not 200 exception is thrown and in this way we do not run into your described issue, but here yeah it seems to be a blocking issue 🤔

Reply:
I agree with Felix that we should not block every failed request.
Other than 400 I see no status code which we should cache as not accepted.